wagtail/admin/forms/choosers.py

```diff
@@ -44,3 +47,90 @@ class EmailLinkChooserForm(forms.Form):
 class PhoneLinkChooserForm(forms.Form):
     phone_number = forms.CharField(required=True)
     link_text = forms.CharField(required=False)
+
+
+class BaseFilterForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_searching = False
+        self.is_filtering_by_collection = False
+        self.search_query = None
+
+    def filter(self, objects):
+        return objects
+
+
+class SearchFilterMixin(forms.Form):
+    """
+    Mixin for a chooser listing filter form, to provide a search field
+    """
+
+    q = forms.CharField(
+        label=_("Search term"),
+        widget=forms.TextInput(attrs={"placeholder": _("Search")}),
+        required=False,
+    )
+
+    def filter(self, objects):
+        objects = super().filter(objects)
+        search_query = self.cleaned_data.get("q")
+        if search_query:
+            search_backend = get_search_backend()
+            objects = search_backend.search(search_query, objects)
+            self.is_searching = True
+            self.search_query = search_query
+        return objects
+
+
+class CollectionFilterMixin(forms.Form):
+    """
+    Mixin for a chooser listing filter form, to provide a collection filter field.
+    The view must pass a `collections` keyword argument when constructing the form
+    """
+
+    def __init__(self, *args, collections=None, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if collections:
+            collection_choices = [
+                ("", _("All collections"))
+            ] + collections.get_indented_choices()
+            self.fields["collection_id"] = forms.ChoiceField(
+                label=_("Collection"),
+                choices=collection_choices,
+                required=False,
+                widget=forms.Select(attrs={"data-chooser-modal-search-filter": True}),
+            )
+
+    def filter(self, objects):
+        collection_id = self.cleaned_data.get("collection_id")
+        if collection_id:
+            self.is_filtering_by_collection = True
+            objects = objects.filter(collection=collection_id)
+        return super().filter(objects)
+
+
+class LocaleFilterMixin(forms.Form):
```

Comment:
Was going to note this was outside the scope of the PR, but it is integral to letting the filter form do its thing.